src/Feature/Redirects/code/Pipelines/RedirectResolver.cs

```diff
@@ -36,9 +36,9 @@ public override void Process(HttpRequestArgs args)
 
         private void Perform301Redirect()
         {
-            var redirectFolder = _contentRepository.GetItem<IRedirectFolder>(new GetItemByQueryOptions
+            var redirectFolder = _contentRepositoryThunk().GetItem<IRedirectFolder>(new GetItemByQueryOptions
```

Comment:
We should avoid calling the accessor function inline like this as it would potentially hide multiple instances if it were configured as transient (which it is). Maybe call it once and assign to a variable?

Reply:
The problem is, you can't store it. The Pipeline processor is only instantiated once for the lifetime of sitecore and then called for each request - so you have to resolve the content repository on each invocation so make sure you get the current context database

Reply:
I just mean that you're calling the think function twice within `Process` which could cause two instances of it to be created. You could just call it once and story it in a locally scoped variable (not a member of the class)

Reply:
Aha, one call is resolving the conte**n**t repository and one the conte**x**t repository! :)

Reply:
Right you are! Sorry for wasting your time there